src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -2461,7 +2461,16 @@ class ShenandoahUpdateHeapRefsTask : public WorkerTask {
               // old-gen region in the most recent collection set, or if this card holds pointers to other non-specific
               // old-gen heap regions.
               if (r->is_humongous()) {
-                r->oop_iterate_humongous(&cl);
+                // Need to examine both dirty and clean cards during mixed evac.
```

Comment:
Sorry, I'm not seeing it. This code looks like it will compute the same `end_of_object` for every region in the humongous object (line 2468), but `start_of_scan` will be `r->bottom`. It seems like this code will have every thread that encounters a region in the humongous object scan the regions between `r` and `end_of_object`. The original method `oop_iterate_humongous` would scan only between `r->bottom` and `r->top`. Maybe we want `end_of_object` to be `r->top` and combine that with only scanning dirty cards?

Reply:
Or, in this case, since we don't care about cards being dirty and we will scan the whole region, could just use the original method?